libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js

```diff
@@ -0,0 +1,141 @@
+const { FolderResourceProvider, ResourceExplorer, FileResource } = require('../lib');
+const path = require('path');
+const assert = require('assert');
+const fs = require('fs');
+const Validator = require('jsonschema').Validator;
+// eslint-disable-next-line security/detect-child-process
+const { exec } = require('child_process');
+
+// Note: This file is intentionally not named *.test.js to ensure it isn't run
+// via `yarn run test` or `npm run test`.
+
+async function runCommand(command) {
+    try {
+        return new Promise((resolve, _reject) => {
+            exec(
+                command,
+                {
+                    cwd: __dirname,
+                },
+                (err, stdout, stderr) => {
+                    if (err || stderr) {
```

Comment:
This is a little unusual - why not use `reject` and `try`/`catch` inside the tests?

Reply:
It's [more or less how .NET does it](https://github.com/microsoft/botbuilder-dotnet/blob/main/tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaTestsFixture.cs#L41). Although there's probably a better way to handle both errors and unmatched schema in the same block: `if (result.error || newSchema !== testsSchema)...`